lib/usb/usb_standard.c

```diff
@@ -277,13 +203,22 @@ usb_standard_set_configuration(usbd_device *usbd_dev,
 	usbd_dev->current_config = found_index + 1;
 
 	if (usbd_dev->current_config > 0) {
-		cfg = &usbd_dev->config[usbd_dev->current_config - 1];
+		cfg = usbd_dev->config[usbd_dev->current_config - 1];
 
 		/* reset all alternate settings configuration */
-		for (i = 0; i < cfg->bNumInterfaces; i++) {
-			if (cfg->interface[i].cur_altsetting) {
-				*cfg->interface[i].cur_altsetting = 0;
+		int totalLen = cfg->wTotalLength;
+		uint8_t *ptr = (uint8_t *)cfg;
+		struct usb_interface_descriptor *iface = 0;
+		while(totalLen > 0) {
+			if(totalLen >= (int)sizeof(struct usb_interface_descriptor)) {
+				if(ptr[1] == USB_DT_INTERFACE) {
+					iface = (struct usb_interface_descriptor *)ptr;
+					if(!usbd_dev->current_iface[iface->bInterfaceNumber])
+						usbd_dev->current_iface[iface->bInterfaceNumber] = iface;
+				}
 			}
+			ptr += ptr[0];
+			totalLen -= ptr[0];
```

Comment:
should not two above lines be swapped?

Reply:
You may be right, but I haven't looked at this patch in a while, and I haven't seen any signs from the maintainers either, so if you want you can make a test case for the problem at line 220, that would clear that up and make it easy for a maintainer to merge the changes as well. But I'm glad someone's inspecting it.

Reply:
you're right, I've fixed the order